smart_contract/executor/derive/src/lib.rs

```diff
@@ -195,3 +199,155 @@ pub fn derive_ref_into_account_owner(input: TokenStream) -> TokenStream {
+/// `iroha_validator::iroha_data_model::evaluate::ExpressionEvaluator`
+/// bound, at least these two should be implemented as well.
+///
+/// Emits a compile error if the struct didn't have all the expected fields with corresponding
+/// types, i.e. `verdict`: `iroha_validator::prelude::Result`, `block_height`: `u64` and
+/// `host`: `iroha_validator::iroha_wasm::Host`, though technically only `verdict` and
+/// `block_height` are needed. The types can be unqualified, but not aliased.
+#[manyhow]
+#[proc_macro_derive(Validate)]
+pub fn derive_validate(input: TokenStream2) -> TokenStream2 {
+    let mut emitter = Emitter::new();
+
+    let Some(input) = emitter.handle(syn2::parse2(input)) else {
+        return emitter.finish_token_stream();
+    };
+
+    let result = default::impl_derive_validate(&mut emitter, &input);
+
+    emitter.finish_token_stream_with(result)
+}
+
+/// Implemets the `iroha_validator::prelude::Visit` trait on a given `Executor` struct.
+/// Users can supply custom overrides for any of the visit functions as freestanding functions
+/// in the same module or omit some of the visit functions completely via the `#[omit(visit(...))]`
+/// attribute by supplying corresponding visit function names inside of it, otherwise a default
+/// implementation from `iroha_validator::default` module is used.
+///
+/// Emits a compile error if the struct didn't have all the expected fields with corresponding
+/// types, i.e. `verdict`: `iroha_validator::prelude::Result`, `block_height`: `u64` and
+/// `host`: `iroha_validator::iroha_wasm::Host`, though technically only `verdict`
+/// is needed. The types can be unqualified, but not aliased.
+///
+/// # Example
+///
+/// ```ignore
+/// use iroha_validator::{permission, prelude::*};
+///
+/// #[derive(Constructor, Entrypoints, ExpressionEvaluator, Validate, Visit)]
+/// #[omit(visit(visit_query)]
+/// pub struct Validator {
+///    verdict: Result,
+///    block_height: u64,
+///    host: iroha_wasm::Host,
+/// }
+///
+/// // Custom visit function should supply a `&mut Validator` as first argument
+/// fn visit_query(validator: &mut Validator, _authority: &AccountId, _query: &QueryBox) {
+///     validator.deny(ValidationFail::NotPermitted(
+///         "All queries are forbidden".to_owned(),
+///     ));
+/// }
+/// ```
+#[manyhow]
+#[proc_macro_derive(Visit, attributes(omit))]
+pub fn derive_visit(input: TokenStream2) -> TokenStream2 {
+    let mut emitter = Emitter::new();
+
+    let Some(input) = emitter.handle(syn2::parse2(input)) else {
+        return emitter.finish_token_stream();
+    };
+
+    let result = default::impl_derive_visit(&mut emitter, &input);
+
+    emitter.finish_token_stream_with(result)
+}
+
+/// Implements three default entrypoints on a given `Executor` struct: `validate_transaction`,
+/// `validate_query` and `validate_instruction`. The `migrate` entrypoint is implied to be
+/// implemented manually by the user if needed.
```

Comment:
`migrate()` entrypoint is mandatory as well as others, so `if needed` doesn't look right to me

Reply:
I see, you don't want to generate `migrate()` because it will be custom in most of the cases. Than how about changing `Entrypoints` to `ValidateEntrypoints`